node/public.go

```diff
@@ -0,0 +1,66 @@
+// Package node provides a high level wrapper around tendermint services.
+package node
+
+import (
+	"fmt"
+
+	"github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/libs/log"
+	"github.com/tendermint/tendermint/libs/service"
+	"github.com/tendermint/tendermint/p2p"
+	"github.com/tendermint/tendermint/privval"
+	"github.com/tendermint/tendermint/proxy"
+	"github.com/tendermint/tendermint/types"
+)
+
+// NewDefault constructs a tendermint node service for use in go process that
+// host their own process-local tendermint node. This is roughly
+// equivalent to running tendermint in it's own process communicating
+// to an external ABCI application.
+func NewDefault(conf *config.Config, logger log.Logger) (service.Service, error) {
+	return newDefaultNode(conf, logger)
+}
+
+// New constructs a tendermint node for use in an integrated
+// context. The ClientCreator makes it possible to construct.
+// The final option is a pointer to a Genesis document: if the value
+// is nil, the genesis document is read from the file specified in the
+// config, and otherwise the node uses value of the final argument.
+func New(conf *config.Config,
+	logger log.Logger,
+	cf proxy.ClientCreator,
+	gen *types.GenesisDoc,
+) (service.Service, error) {
```

Comment:
What is the main reason to going with service.Service here?

Reply:
I don't really like the service.Service interface, but it's pervasive in the code base, and functionally describes the way that *most* not invasive users of node interact with the service. I think at some point we can rethink this interface (perhaps during 0.36?) but for now I think it's best to sort of use this type as the first step in encapsulating the Node.

Reply:
I was also going to ask you what you think about `service.Service`. I was just looking at it before and I wasn't completely convinced by the implementation so I wondered if you had plans to change it

Reply:
I think it provides a clean interface for callers to use, i.e. to start/stop.

Reply:
We could also expose a subset of the `Service` interface.

Reply:
for sure! I think that's definitely a conversation that we should have. 

we already export service, so it seemed obvious at this stage.